src/scripts/main.js

```diff
@@ -2,10 +2,23 @@
 
 function waitFor(element, eventName) {
   // write your code here
+  return new Promise((resolve, reject) => {
+    element.addEventListener(eventName, () => {
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}`,
+      );
+    });
+  });
```

Comment:
The `waitFor` function resolves the promise the first time the event is triggered. If you want to handle multiple events of the same type on the same element, you might need to reconsider this approach, as the promise will only resolve once per event type per element.